financeager/period.py

```diff
@@ -57,6 +58,89 @@ def year(self):
         """Return period year as integer."""
         return int(self._name)
 
+    @abstractmethod
+    def add_entry(self, table_name=None, **kwargs):
+        """
+        Add an entry (standard or recurrent) to the database.
+        If 'table_name' is not specified, the kwargs name, value[, category,
+        date] are used to insert a unique entry in the standard table.
+        With 'table_name' as 'recurrent', the kwargs name, value, frequency
+        [, start, end, category] are used to insert a template entry in the
+        recurrent table.
+        Two kwargs are mandatory:
+            :param name: entry name
+            :type name: str
+            :param value: entry value
+            :type value: float, int or str
+        The following kwarg is optional:
+            :param category: entry category. If not specified, the program
+                attempts to derive it from previous, eponymous entries. If this
+                fails, ``_DEFAULT_CATEGORY`` is assigned
+            :type category: str or None
+        The following kwarg is optional for standard entries:
+            :param date: entry date. Defaults to current date
+            :type date: str of ``PERIOD_DATE_FORMAT``
+        The following kwarg is mandatory for recurrent entries:
+            :param frequency: 'yearly', 'half-yearly', 'quarter-yearly',
+                'bimonthly', 'monthly', 'weekly' or 'daily'
+        The following kwargs are optional for recurrent entries:
+            :param start: start date (defaults to current date)
+            :param end: end date (defaults to last day of the period's year)
+        :raise: PeriodException if validation failed or table name unknown
+        :return: ID of new entry (int)
+        """
+        pass
```

Comment:
You can remove all the `pass`. Having a docstring alone already is a syntactically valid way to define a class.